Core/clim-core/panes/layout-protocol.lisp

```diff
@@ -473,13 +469,30 @@ which changed during the current execution of CHANGING-SPACE-REQUIREMENTS.
 
 (defmethod change-space-requirements ((pane layout-protocol-mixin)
                                       &key resize-frame &allow-other-keys)
-  (when-let ((parent (sheet-parent pane)))
-    (change-space-requirements parent :resize-frame resize-frame)))
+  ;; do nothing here
+  nil)
 
 (defmethod change-space-requirements :after ((pane layout-protocol-mixin)
                                              &key resize-frame &allow-other-keys)
```

Comment:
this logic for the top-level-sheet-pane should be implemented in the top-level-sheet-pane *primary* method. Notice, that doing that in the after method violates the changing-space-requirements macro specification.

Reply:
I don't understant but maybe is the same prolem of above.

Reply:
all calls to change-space-requirements which appear in a dynamic scope of `changing-space-requirements` should be simply recorded for later execution, while this after method triggers a traversal of a whole tree eagerly. Our implementation of `changing-space-requirements` is wrong either way, so I suppose that could be fixed /then/. It is not related to the previous question.

That said, please provide an example code that illustrates the issue with menus and explain what in the current implementation of menus makes it non-viable to call layout-frame directly (when resize-frame is given).

Reply:
It is the `layout-frame` call that its recorded for later execution.
I think that the most straight way will be something like:
```
(defmethod change-space-requirements :after ((pane layout-protocol-mixin)
                                             &key resize-frame &allow-other-keys)
  (when-let ((parent (sheet-parent pane)))
    (if resize-frame
        (if *changing-space-requirements*
            (unless (find pane *changed-space-requirements* :key #'second)
              (push (list (pane-frame pane) pane resize-frame)
                    *changed-space-requirements*))
            (layout-frame (pane-frame pane)))
        (note-space-requirements-changed parent pane))))
```
but we can't because `pane-frame` when its called on a menu pane (like Listener completion menu or presentation menu- right click-) return the main frame and not the menu-frame. So it will be the main frame and not the menu to be resized.